core/vm/instructions.go

```diff
@@ -514,25 +513,18 @@ func opCreate(instr instruction, pc *uint64, env Environment, contract *Contract
 		offset, size = stack.pop(), stack.pop()
 		input        = memory.Get(offset.Int64(), size.Int64())
 		gas          = new(big.Int).Set(contract.Gas)
-		addr         common.Address
-		ret          []byte
-		suberr       error
 	)
-
 	contract.UseGas(contract.Gas)
-	ret, addr, suberr = env.Create(contract, input, gas, contract.Price, value)
-	if suberr != nil {
+	_, addr, suberr := env.Create(contract, input, gas, contract.Price, value)
+	// Push new item to the stack based on the returned error (and phase)
+	// If everything was successfull and no error was returned the address
+	// of the created contract is pushed to the stack, otherwise 0.
+	if params.IsHomestead(env.BlockNumber()) && suberr == CodeStoreOutOfGasError {
+		stack.push(new(big.Int))
+	} else if suberr != nil && suberr != CodeStoreOutOfGasError {
```

Comment:
Sure, that is fine, but then could you add a comment to emphasize that this is by design and not an omission? The current code "looks" buggy, even if it's actually correct :)